cpp/src/parquet/arrow/schema_internal.cc

```diff
@@ -191,10 +191,18 @@ Result<std::shared_ptr<ArrowType>> FromInt64(const LogicalType& logical_type) {
 Result<std::shared_ptr<ArrowType>> GetArrowType(
     Type::type physical_type, const LogicalType& logical_type, int type_length,
     const ArrowReaderProperties& reader_properties) {
-  if (logical_type.is_invalid() || logical_type.is_null()) {
+  if (logical_type.is_null()) {
     return ::arrow::null();
   }
 
+  if (logical_type.is_invalid() && reader_properties.convert_unknown_logical_types()) {
+    return GetArrowType(physical_type, *NoLogicalType::Make(), type_length,
+                        reader_properties);
+  } else if (logical_type.is_invalid()) {
+    return Status::NotImplemented(
```

Comment:
Is this a behavior change which returns `arrow::null()` in the past but now fails?